modFastVM/src/org/aion/fastvm/Callback.java

```diff
@@ -173,7 +162,21 @@ public static void putStorage(byte[] address, byte[] key, byte[] value) {
         // System.err.println("PUT_STORAGE: address = " + Hex.toHexString(address) + ", key = " +
         // Hex.toHexString(key) + ", value = " + Hex.toHexString(value));
 
-        repo().addStorageRow(Address.wrap(address), new DataWord(key), new DataWord(value));
+        if (value == null || value.length == 0 || isZero(value)) {
+            kernelRepo().removeStorage(AionAddress.wrap(address), key);
+        } else {
+            kernelRepo().putStorage(AionAddress.wrap(address), key, value);
+        }
+    }
+
+    private static boolean isZero(byte[] value) {
+        int length = value.length;
+        for (int i = 0; i < length; i++) {
+            if (value[length - 1 - i] != 0) {
```

Comment:
why just use
for (byte b : value) {
    if (b != 0) {
        return false;
    }
}

Reply:
see my response to kelvin above, he brought up the same issue